### PR TITLE
deleter now deletes files in addition to directories

### PR DESCRIPTION
**Pull request by lspahija**
**Description**

[deleter.py](https://github.com/commaai/openpilot/blob/master/system/loggerd/deleter.py) only deleted directories in `Paths.log_root()` and ignored ordinary top-level files. Because it ignored files, [this bug was reported](https://github.com/commaai/openpilot/issues/30102). 

The code now first deletes ordinary files, followed by directories that are not openpilot segments, followed by openpilot segment directories according to the existing ordering from oldest to newest.

**Verification**

Unit tests were added to `system/loggerd/tests/test_deleter.py` that failed prior to the bugfix being implemented. After the bugfix was implemented, these tests now pass.

**Note**

Please note that this bugfix carries a [$100 bounty](https://github.com/commaai/openpilot/issues/30102).

### CODE REVIEW

File: system/loggerd/deleter.py [hunk 1]
```diff
@@ -52,15 +73,19 @@ def deleter_thread(exit_event):
 
     if out_of_percent or out_of_bytes:
       dirs = listdir_by_creation(Paths.log_root())
+      segments, non_segments = separate_segments_and_nonsegments(dirs)
```
Comment (adeebshihadeh) [inline system/loggerd/deleter.py:76]: it's nice that deleter doesn't really know much about "segments". let's keep it that way

File: system/loggerd/tests/test_deleter.py [hunk 2]
```diff
@@ -73,6 +80,39 @@ def test_delete_order(self):
       self.make_file_with_data(self.seg_format2.format(0), self.f_type),
     ])
 
+  def test_delete_both_files_and_dirs(self):
```
Comment (adeebshihadeh) [inline system/loggerd/tests/test_deleter.py:83]: let's also create lots of random nested dirs and files to really cover everything

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]

File: system/loggerd/deleter.py (hunk 1 above)
Comment (lspahija) [reply system/loggerd/deleter.py]: done

File: system/loggerd/tests/test_deleter.py (hunk 2 above)
Comment (lspahija) [reply system/loggerd/tests/test_deleter.py:114]: now generating random nested dirs and files

**Comment by lspahija** [comment]
@adeebshihadeh re-review?

**Comment by adeebshihadeh** [comment]
Closing. This is the second time you've asked for a review, and this is just not good enough. There's unused code, and it even calls `os.listdir` twice.

**Pull request closed without merging**